EmployeeLeaveTracking/EmployeeLeaveTracking.Data/DTOs/DesignationMasterDTO.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace EmployeeLeaveTracking.Data.DTOs
+{
+    public class DesignationMasterDTO
+    {
+        public int Id { get; set; }
+
+
+        [Required(ErrorMessage = "Designation type name is a required field.")]
+        [MaxLength(50, ErrorMessage = "Maximum length for the designation type name is 50 characters.")]
+        public string? DesignationName { get; set; }
```

Comment:
Just out of curiosity, If we are applying Required annotations then in which scenario we are expecting a null value for DesignationName prop

Reply:
Removed the Contradictory Logic